File: cs214/projects/proj10/ruby/Penguin.rb
# Penguin.rb | Defines the Penguin class which inherits attributes and methods
#   from the WalkingBird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Advait Scaria
# Date: 04/27/2019
####################################################

require './Bird.rb'

class Penguin < WalkingBird

  ####################################################
  # A Penguin's Call
  # Return: a Penguin sound ("Huh-huh-huh-huuuuh").
  ####################################################
  def call
    "Huh-huh-huh-huuuuh"
  end

end

File: cs214/projects/proj10/java/Penguin.java
/* Penguin.java provides a Penguin subclass of WalkingBird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Advait Scaria
 * Date: 04/27/2019
 ******************************************************/

public class Penguin extends WalkingBird {

 /* default constructor
  * PostCond: invokes superclass Bird's 
  *     	  default constructor to set
  *			  myName = "".
  */ 
  public Penguin() {
      super();
  }

 /* explicit constructor
  * Receive: name, a String 
  * PostCond: myName == name.
  */
  public Penguin(String name)
  {     
      super(name);
  }

 /* A Penguin's Call
  * Return: a penguin-call ("Huh-huh-huh-huuuuh").
  */
  public String call()
  {
     return "Huh-huh-huh-huuuuh";
  }

} 

